duckduckgo/Android
# Add long press context menu for images with option to download image

Asana Issue URL: https://app.asana.com/0/414730916066338/538612605030335

## Description
Add ability to long press on an image in a WebView and see a context menu allowing for the image to be downloaded

Note, images won't be able to be downloaded from DDG SERP yet due to an ongoing issue there.

## Steps to Test this PR:
1. Visit a site which contains images, such as http://www.bbc.co.uk/news/business-42957834
1. Check that you can long press on an image (such as the FTSE 100 chart 📉) and see an option to download the image. Do so.
1. Verify you are prompted for the runtime permission for writing to storage.
1. Deny the permission, and ensure the app handles it gracefully enough.
1. Try again, this time grant the permission, and ensure the image downloads successfully.
1. Check that you cannot long press on the video on the page.
1. Check that you can long press on the `BBC` icon (very top left of the page) and that you can download this image - this is a different type of image from the FTSE chart but both should download successfully.
1. Also, check an image from theguardian.com as these images were previously not being named sensibly - it should name them sensibly now.

From here, it would be wise to revoke the storage permission, and test that PDFs still download successfully (using espc.com for instance, and clicking on a property, and then "schedule" to trigger a PDF download)
